library/DynamicReplicatedTableEntity.cs

```diff
@@ -163,61 +168,89 @@ internal DynamicReplicatedTableEntity(string partitionKey, string rowKey, DateTi
+        /// Deserializes this <see cref="DynamicTableEntity"/> instance using the specified <see cref="Dictionary{TKey,TValue}"/> of property names to values of type <see cref="object"/>.
         /// </summary>
-        /// <param name="properties">A collection containing the <see cref="Dictionary{TKey,TValue}"/> of string property names mapped to values of type <see cref="EntityProperty"/> to store in this <see cref="DynamicTableEntity"/> instance.</param>
+        /// <param name="properties">A collection containing the <see cref="Dictionary{TKey,TValue}"/> of string property names mapped to values of type <see cref="object"/> to store in this <see cref="DynamicTableEntity"/> instance.</param>
         /// <param name="operationContext">An <see cref="OperationContext"/> object used to track the execution of the operation.</param>
-        public override void ReadEntity(IDictionary<string, EntityProperty> properties, OperationContext operationContext)
+        public override void ReadEntity(IDictionary<string, object> properties)
         {
-            Dictionary<string, EntityProperty> prop = new Dictionary<string, EntityProperty>(properties);
+            Dictionary<string, object> prop = new Dictionary<string, object>(properties);
 
             // Read ReplicatedTable meta data
-            _rtable_RowLock = (bool)prop["_rtable_RowLock"].BooleanValue; prop.Remove("_rtable_RowLock");
-            _rtable_Version = (long)prop["_rtable_Version"].Int64Value; prop.Remove("_rtable_Version");
-            _rtable_Tombstone = (bool)prop["_rtable_Tombstone"].BooleanValue; prop.Remove("_rtable_Tombstone");
-            _rtable_ViewId = (long)prop["_rtable_ViewId"].Int64Value; prop.Remove("_rtable_ViewId");
-            _rtable_Operation = (string)prop["_rtable_Operation"].StringValue; prop.Remove("_rtable_Operation");
-            _rtable_BatchId = (Guid)prop["_rtable_BatchId"].GuidValue; prop.Remove("_rtable_BatchId");
-            _rtable_LockAcquisition = (DateTimeOffset)prop["_rtable_LockAcquisition"].DateTimeOffsetValue; prop.Remove("_rtable_LockAcquisition");
+            _rtable_RowLock = (bool)prop["_rtable_RowLock"]; prop.Remove("_rtable_RowLock");
+            _rtable_Version = (long)prop["_rtable_Version"]; prop.Remove("_rtable_Version");
+            _rtable_Tombstone = (bool)prop["_rtable_Tombstone"]; prop.Remove("_rtable_Tombstone");
+            _rtable_ViewId = (long)prop["_rtable_ViewId"]; prop.Remove("_rtable_ViewId");
+            _rtable_Operation = (string)prop["_rtable_Operation"]; prop.Remove("_rtable_Operation");
+            _rtable_BatchId = (Guid)prop["_rtable_BatchId"]; prop.Remove("_rtable_BatchId");
+            _rtable_LockAcquisition = (DateTimeOffset)prop["_rtable_LockAcquisition"]; prop.Remove("_rtable_LockAcquisition");
+            prop.Remove("PartitionKey");
```

Comment:
I think we should keep these properties and may be fix the tests to account for this ?